sys/net/gnrc/network_layer/ipv6/ext/gnrc_ipv6_ext.c

```diff
@@ -26,52 +29,61 @@
 #include "debug.h"
 
 #ifdef MODULE_GNRC_IPV6_EXT_RH
-static int _handle_rh(gnrc_pktsnip_t *current, gnrc_pktsnip_t *pkt)
+static void _forward_pkt(gnrc_pktsnip_t *pkt)
+{
+    gnrc_pktsnip_t *netif_snip, *ipv6_snip;
+    ipv6_hdr_t *ipv6;
+
+    ipv6_snip = gnrc_pktsnip_search_type(pkt, GNRC_NETTYPE_IPV6);
+    /* there should be an IPv6 header */
+    assert(ipv6_snip != NULL);
+    ipv6 = ipv6_snip->data;
+    if (--(ipv6->hl) == 0) {
+        DEBUG("ipv6_ext_rh: hop limit reached 0: drop packet\n");
+        gnrc_pktbuf_release(pkt);
+    }
+    /* remove L2 headers around IPV6 */
+    netif_snip = gnrc_pktsnip_search_type(pkt, GNRC_NETTYPE_NETIF);
+    if (netif_snip != NULL) {
+        gnrc_pktbuf_remove_snip(pkt, netif_snip);
+    }
+    pkt = gnrc_pktbuf_reverse_snips(pkt);
+    if (pkt == NULL) {
+        DEBUG("ipv6_ext_rh: can't reverse snip order in packet");
+        return;
+    }
+    /* forward packet */
+    if (!gnrc_netapi_dispatch_send(GNRC_NETTYPE_IPV6,
+                                   GNRC_NETREG_DEMUX_CTX_ALL,
+                                   pkt)) {
+        DEBUG("ipv6_ext_rh: could not dispatch packet to the IPv6 "
+              "thread\n");
+        gnrc_pktbuf_release(pkt);
+    }
+}
+
+static int _handle_rh(gnrc_pktsnip_t *pkt)
 {
     gnrc_pktsnip_t *ipv6;
-    ipv6_ext_t *ext = (ipv6_ext_t *) current->data;
-    size_t current_offset;
+    ipv6_ext_t *ext = (ipv6_ext_t *)pkt->data;
     ipv6_hdr_t *hdr;
     int res;
 
-    /* check seg_left early to avoid duplicating the packet */
+    /* check seg_left early to avoid to exit quickly */
```

Comment:
Did you mean "to exit quickly" instead of "to avoid to exit quickly"?

Reply:
Yepp :-)